### PR TITLE
style: Use SmallVec more in the style system.

### PR DESCRIPTION
**Pull request by emilio**
I expect this to improve the situation significantly in https://bugzilla.mozilla.org/show_bug.cgi?id=1360881

This is worth doing on its own, though.

---
This change is [Reviewable](https://reviewable.io/reviews/servo/servo/16661)

### CODE REVIEW

**Comment by emilio** [comment]
I ended up only doing this for computed values, since we care a lot about the size of specified ones.

File: components/style/properties/helpers/animated_properties.mako.rs [hunk 1]
```diff
@@ -622,6 +623,16 @@ impl<T: RepeatableListInterpolate> Interpolate for Vec<T> {
...
 }
 
+impl<T: RepeatableListInterpolate> Interpolate for SmallVec<[T; 1]> {
+    fn interpolate(&self, other: &Self, progress: f64) -> Result<Self, ()> {
+        use num_integer::lcm;
+        let len = lcm(self.len(), other.len());
+        self.iter().cycle().zip(other.iter().cycle()).take(len).map(|(me, you)| {
+            me.interpolate(you, progress)
+        }).collect()
+    }
```
Comment (bholley) [inline components/style/properties/helpers/animated_properties.mako.rs]: Not wild about this duplication. Can you implement this for anything that's VecLike + has an iter() method? VecLike should really require iter() IMO, but making a breaking change to smallvec would be annoying, so you can just add a trait here.

File: components/style/properties/helpers/animated_properties.mako.rs [hunk 2]
```diff
@@ -2213,6 +2229,27 @@ impl<T: ComputeDistance> ComputeDistance for Vec<T> {
     }
 }
 
+impl<T: ComputeDistance> ComputeDistance for SmallVec<[T; 1]> {
```
Comment (bholley) [inline components/style/properties/helpers/animated_properties.mako.rs]: Same here.

File: components/style/values/computed/mod.rs [hunk 3]
```diff
@@ -332,7 +332,7 @@ pub struct Shadow {
 /// A `<number>` value.
 pub type Number = CSSFloat;
 
-#[derive(Debug, Clone, PartialEq)]
+#[derive(Debug, Copy, Clone, PartialEq)]
```
Comment (bholley) [inline components/style/values/computed/mod.rs]: Not totally sure about this, but I guess it's fine?

**Comment by bholley** [comment]
r=me with the duplication removed. Thanks for doing this!

@bors-servo delegate+

File: components/style/values/computed/mod.rs (hunk 3 above)
Comment (emilio) [inline components/style/values/computed/mod.rs]: It's in `values::Computed`, so it's just a discriminant + one float. We definitely derive `Copy` on bigger stuff :)

**Comment by emilio** [comment]
@bors-servo r=bholley

 * turns out we don't need the impl for `Vec` after I removed the text shadow stuff away from it, so I'll let other people to parameterize over `ExactSizeIterator<T>` if/when needed.

**Pull request merged**